distribution/build.gradle

```diff
@@ -196,4 +297,76 @@ DependencyLicensesTask.configure(project) {
   mapping from: /jackson-.*/, to: 'jackson'
 }
 
-task run(type:org.elasticsearch.gradle.test.RunTask){}
\ No newline at end of file
+task run(type:org.elasticsearch.gradle.test.RunTask){}
+
+/**
+ * Build some variables that are replaced in the packages. This includes both
+ * scripts like bin/elasticsearch and bin/plugin that a user might run and also
+ * scripts like postinst which are run as part of the installation.
+ *
+ * <dl>
+ *  <dt>package.name</dt>
+ *  <dd>The name of the project. Its sprikled throughout the scripts.</dd>
+ *  <dt>package.version</dt>
+ *  <dd>The version of the proejct. Its mostly used to find the exact jar name.
+ *    </dt>
+ *  <dt>path.conf</dt>
+ *  <dd>The default directory from which to load configuration. This is used in
+ *    the packaging scripts, but in that context it is always
+ *    /etc/elasticsearch. Its also used in bin/plugin, where it is
+ *    /etc/elasticsearch for the os packages but $ESHOME/config otherwise.</dd>
+ *  <dt>path.env</dt>
+ *  <dd>The env file sourced before bin/elasticsearch to set environment
+ *    variables. Think /etc/defaults/elasticsearch.</dd>
+ *  <dt>heap.min and heap.max</dt>
+ *  <dd>Default min and max heap</dd>
+ *  <dt>scripts.footer</dt>
+ *  <dd>Footer appended to control scripts embedded in the distribution that is
+ *    (almost) entirely there for cosmetic reasons.</dd>
+ *  <dt>stopping.timeout</dt>
+ *  <dd>RPM's init script needs to wait for elasticsearch to stop before
+ *    returning from stop and it needs a maximum time to wait. This is it. One
+ *    day. DEB retries forever.</dd>
+ * </dl>
+ */
+def expansionsForDistribution(distributionType) {
+  def footer = "# Built for ${project.name}-${project.version} " +
+      "(${distributionType})"
+  def expansions = [
+    'project.name': project.name,
+    'project.version': version,
+
+    'path.conf': [
+      'tar': '$ES_HOME/config',
+      'zip': '$ES_HOME/config',
+      'def': '/etc/elasticsearch',
+    ],
+    'path.env': [
+      'deb': '/etc/default/elasticsearch',
+      'rpm': '/etc/sysconfig/elasticsearch',
+      /* There isn't one of these files for tar or zip but its important to
+        make an empty string here so the script can properly skip it. */
+      'def': '',
+    ],
+
+    'heap.min': '256m',
+    'heap.max': '1g',
+
+    'stopping.timeout': [
+      'rpm': 86400,
+    ],
+
+    'scripts.footer': [
+      /* Debian needs exit 0 on these scripts so we add it here and preserve
+        the pretty footer. */
+      'deb': "exit 0\n$footer",
+      'def': footer
+    ],
+  ]
+  return expansions.collectEntries { key, value ->
+    if (value instanceof Map) {
+      value = value[distributionType] ?: value['def'] // def is for default!
```

Comment:
What does this do in the case of eg stopping.timeout where there is no default, it is only for rpm?